crates/crypto/src/key.rs

```diff
@@ -0,0 +1,182 @@
+use zeroize::ZeroizeOnDrop;
+
+use crate::{AlgorithmInstance, SignatureError, VerificationError};
+
+#[derive(Debug, thiserror::Error)]
+#[error("invalid public key")]
+pub struct InvalidPublicKey;
+
+/// Public key.
+#[non_exhaustive]
+pub enum PublicKey {
```

Comment:
Was it necessary to start refactoring `ssi_crypto` in the PR? Now we have multiple keys/signing/verification implementations, and depending on the credential format it will use one of the other